File: LeetCode/2196-create-binary-tree-from-descriptions/2196-create-binary-tree-from-descriptions.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode createBinaryTree(int[][] descriptions) {
        Map<Integer, TreeNode> mp = new HashMap<>();
        Set<Integer> childSet = new HashSet<>();
        for (int[] x : descriptions) {
            int p = x[0];
            int c = x[1];
            int l = x[2];
            mp.putIfAbsent(p, new TreeNode(p));
            mp.putIfAbsent(c, new TreeNode(c));
            if (l == 1) {
                mp.get(p).left = mp.get(c);
            } else {
                mp.get(p).right = mp.get(c);
            }
            childSet.add(c);
        }
        for (int[] x : descriptions) {
            int p = x[0];
            if (!childSet.contains(p)) {
                return mp.get(p);
            }
        }
        return null;
    }
}

File: LeetCode/2196-create-binary-tree-from-descriptions/2196-create-binary-tree-from-descriptions.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def createBinaryTree(self, descriptions: List[List[int]]) -> Optional[TreeNode]:
        mp = {}
        childSet = set()
        
        for x in descriptions:
            p, c, l = x
            if p not in mp:
                mp[p] = TreeNode(p)
            if c not in mp:
                mp[c] = TreeNode(c)
                
            if l == 1:
                mp[p].left = mp[c]
            else:
                mp[p].right = mp[c]
                
            childSet.add(c)
            
        for x in descriptions:
            p = x[0]
            if p not in childSet:
                return mp[p]
                
        return None